src/libAtomVM/defaultatoms.c

```diff
@@ -23,10 +23,15 @@
 #include <stdlib.h>
 #include <string.h>
 
+// About X macro: https://en.wikipedia.org/wiki/X_macro
+#define X(name, lenstr, str) \
+    _Static_assert(*lenstr == strlen(str), "Macro length mismatch for " # name);
```

Comment:
I was waiting this check for quite some time, let's enable it only with clang >= 17. That will be enough at least for blocking any PR that introduces some atom length bug.